packages/jest-circus/src/eventHandler.ts

```diff
@@ -88,8 +88,15 @@ const eventHandler: Circus.EventHandler = (
       break;
     }
     case 'add_hook': {
-      const {currentDescribeBlock, hasStarted} = state;
+      const {currentDescribeBlock, currentlyRunningTest, hasStarted} = state;
       const {asyncError, fn, hookType: type, timeout} = event;
+
+      if (currentlyRunningTest) {
+        throw new Error(
```

Comment:
@jeysal is it better to push into `unhandledErrors` like you do below rather than throwing?

Reply:
I think so, yes. IIUC:

1. Nowhere else in the event handler (other than the places where you recently added it) throws (so just for consistency even if the result was the same).
2. It will allow circus to collect another error if the test itself also throws and give both to the reporter (this is the main difference I think).

I'd argue change it (and the other occurrence that currently throws).

Reply:
Can have an `else` for the other, less specific errors

Reply:
good stuff, thanks!